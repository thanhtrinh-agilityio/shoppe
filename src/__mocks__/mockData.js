export const PRODUCT_ITEM = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: 'Sold out',
}

export const PRODUCT_DETAIL = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: 'Sold out',
  quantity: 1,
  introduction:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
}

export const LIST_PRODUCT = [
  {
    id: 1,
    imageUrl: '/images/liraEarring.svg',
    name: 'Lira Earrings',
    price: 20,
    status: '-25%',
  },
  {
    id: 2,
    imageUrl: '/images/liraEarring.svg',
    name: 'Lira Earrings',
    price: 20,
    status: '',
  },
  {
    id: 3,
    imageUrl: '/images/liraEarring.svg',
    name: 'Kaede Hair Pin Set Of 3',
    price: 20,
    status: '',
  },
  {
    id: 4,
    imageUrl: '/images/liraEarring.svg',
    name: 'Hair Pin Set of 3',
    price: 20,
  },
  {
    id: 5,
    imageUrl: '/images/liraEarring.svg',
    name: 'Plaine Necklace',
    price: 20,
    status: 'Sold out',
  },
  {
    id: 6,
    imageUrl: '/images/liraEarring.svg',
    name: 'Yuki Hair Pin Set of 3',
    price: 20,
    status: '',
  },
]
