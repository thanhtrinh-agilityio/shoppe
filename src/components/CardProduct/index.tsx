import { memo } from 'react'
import Link from 'next/link'
import Image, { ImageLoader } from 'next/image'
import { Box, Heading, Text } from '@chakra-ui/react'

// Types
import { IProduct } from '@self-types/index'

// Utils
import { currencyFormat } from '@utils/index'

// Services
import myImageLoader from '@services/imageLoader'

interface CardProductProps {
  product: IProduct
}

const CardProduct = ({ product }: CardProductProps) => {
  const { imageUrl, status, name, price, id } = product
  return (
    <Box maxW="sm" borderRadius="lg" overflow="hidden" position="relative">
      {status && (
        <Box
          position="absolute"
          top="16px"
          left="16px"
          background="beaver"
          borderRadius="4px"
          textAlign="center"
          padding="3px 6px"
        >
          <Text fontSize="12px" textColor="light">
            {status}
          </Text>
        </Box>
      )}
      <Link href={`/products/${id}`} key={id}>
        <Image
          width={300}
          height={300}
          src={imageUrl}
          alt={name}
          loader={myImageLoader as ImageLoader}
        />
      </Link>
      <Box pt="24px" color="dark" fontSize="medium">
        <Heading as="h3" fontSize="medium" fontWeight="base" color="dark">
          <Link href={`/products/${id}`}>{name}</Link>
        </Heading>
        <Text mt="16px" textColor="beaver">
          {currencyFormat(price)}
        </Text>
      </Box>
    </Box>
  )
}

export default memo(CardProduct)
