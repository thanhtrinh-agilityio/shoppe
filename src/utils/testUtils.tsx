import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
