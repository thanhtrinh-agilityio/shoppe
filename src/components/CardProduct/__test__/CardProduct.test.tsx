// Mocks
import { PRODUCT_ITEM } from '@mocks/mockData'

// Utils
import { render, screen } from '@utils/testUtils'

// Components
import CardProduct from '@components/CardProduct'

describe('CardProduct render', () => {
  it('Should match data for card component', () => {
    const { container } = render(<CardProduct product={PRODUCT_ITEM} />)

    expect(container).toMatchSnapshot()
  })

  it('Should navigate to product details page', () => {
    render(<CardProduct product={PRODUCT_ITEM} />)

    const link = screen.getByRole('link')

    expect(link.getAttribute('href')).toEqual(`/products/${PRODUCT_ITEM.id}`)
    expect(link.textContent).toBe(PRODUCT_ITEM.name)
  })
})
