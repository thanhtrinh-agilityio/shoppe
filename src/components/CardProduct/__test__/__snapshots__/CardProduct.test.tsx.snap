// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardProduct render Should match data for card component 1`] = `
<div>
  <div
    class="css-7ccns3"
  >
    <div
      class="css-ervdjc"
    >
      <p
        class="chakra-text css-1ok93o9"
      >
        Sold out
      </p>
    </div>
    <img
      alt="Lira Earrings"
      data-nimg="1"
      decoding="async"
      height="300"
      loading="lazy"
      src="https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042?w=640&q=75"
      srcset="https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042?w=384&q=75 1x, https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042?w=640&q=75 2x"
      style="color: transparent;"
      width="300"
    />
    <div
      class="css-1xzknfv"
    >
      <h3
        class="chakra-heading css-7ex3dd"
      >
        <a
          href="/products/1"
        >
          Lira Earrings
        </a>
      </h3>
      <p
        class="chakra-text css-1ovs4to"
      >
        $ 20.00
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
